cernopendata/mappings/v7/records/record-v1.0.0.json

```diff
@@ -0,0 +1,166 @@
+{
+  "mappings": {
```

Comment:
Perhaps you can remove v5 mappings now that we have v7 everywhere?

Reply:
I can remove it. I wanted to know if it is working fine on your side as well. If everything is good to go, I will remove it.

Reply:
It does not actually work out of the box. I cannot load records, as the ES container keeps restarting, with messages like:

```console 
$ docker-compose -f docker-compose-dev.yml down -v
$  docker-compose -f docker-compose-dev.yml build
$ docker-compose -f docker-compose-dev.yml up -d
$ docker exec -i -t opendatacernch_web_1 /code/scripts/populate-instance.sh --skip-records
...
$ docker logs opendatacernch_elasticsearch_1 
...
ERROR: [3] bootstrap checks failed
[1]: memory locking requested for elasticsearch process but memory is not locked
[2]: max virtual memory areas vm.max_map_count [65530] is too low, increase to at least [262144]
[3]: the default discovery settings are unsuitable for production use; at least one of [discovery.seed_hosts, discovery.seed_providers, cluster.initial_master_nodes] must be configured
ERROR: Elasticsearch did not exit normally - check the logs at /usr/share/elasticsearch/logs/docker-cluster.log
```

Seems like we need to set some new params that are not appearing in this PR's `docker-compose-dev.yml`?

Reply:
Tested loading full site, and things mostly work, but I see some discrepancies comparing against production, e.g. see the total count:

```
/api/records

- localhost: "total": 7621
- DEV: "total": 5109
```

Some collections seem OK, for example:

```
/api/records?q=alice

- localhost; "total": 35
- DEV; "total": 35

/api/records?q=experiment:opera

- localhost; "total": 910
- DEV; "total": 910
```

The extra count seems to come from CMS records:

```
/api/records?q=experiment:cms

- localhost; "total": 6506
- DEV; "total": 3994
```

I haven't looked more deeply to see what is different. Perhaps we need to amend fixture loading or something... Can you please have a look?

P.S. If we need Elasticsearch 7 to unblock the upcoming work on the search UI adaptations (CC @mvidalgarcia), then we could merge "as is" and create a new issue for the "extra count" observations.

Reply:
OK, mystery understood :smile: The extra records (7621 vs 5109) are coming from "on-demand datasets" that we usually don't index by default in the QA and PROD and that we show in the results only when the corresponding "include on-demand datasets" check box in the UI is ticked. See https://github.com/cernopendata/opendata.cern.ch/issues/2919#issuecomment-734362529

Hence we can finalise this PR as is and address the on-demand parts later as part of the search UI and facets.